sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/cli/XrayWrapperArtifactoryClientSupport.java

```diff
@@ -0,0 +1,257 @@
+
+    XrayWrapperReportSupport reportReader;
+
+    public enum ScanStatus {
+        SCANNED("scanned"),
+
+        IN_PROGRESS("in progress"),
+
+        NOT_SCANNED("not scanned"),
+
+        UNKNOWN("unknown");
+
+        private String statusValue;
+
+        ScanStatus(String status) {
+            this.statusValue = status.toLowerCase();
+        }
+
+        public String getStatusValue() {
+            return this.statusValue;
+        }
+
+        public static ScanStatus fromString(String stringStatus) throws XrayWrapperException {
+            if (stringStatus == null) {
+                throw new XrayWrapperException("Scan status is NULL", XrayWrapperExitCode.UNKNOWN_ERROR);
+            }
+            for (ScanStatus status : ScanStatus.values()) {
+                if (status.statusValue.equals(stringStatus.toLowerCase())) {
+                    return status;
+                }
+            }
+            throw new XrayWrapperException("Received unexpected scan status", XrayWrapperExitCode.UNSUPPORTED_API_REQUEST);
+        }
+    }
+
+    public XrayWrapperArtifactoryClientSupport(XrayWrapperConfiguration xrayWrapperConfiguration, XrayWrapperArtifact artifact) {
+        this.xrayWrapperConfiguration = xrayWrapperConfiguration;
+        this.artifactoryClient = new XrayAPIArtifactoryClient(artifact, xrayWrapperConfiguration);
+        this.reportReader = new XrayWrapperReportSupport();
+    }
+
+    /**
+     * Controls which http requests are send to the xray artifactory, downloads the
+     * reports after a successful scan and add security information from Security
+     * report to the cycloneDX report further the artifacts are deleted from the
+     * artifactory
+     *
+     * @throws XrayWrapperException
+     */
+    public void waitForScansToFinishAndDownloadReport() throws XrayWrapperException {
+        ClientSupportContext context = new ClientSupportContext();
+
+        // get xray version from artifactory
+        String xray_version = artifactoryClient.requestXrayVersion();
+        LOG.debug("Artifactory available, scan with Xray version: {}", xray_version);
+
+        // check if artifact is uploaded
+        artifactoryClient.checkArtifactoryUploadSuccess();
+        LOG.debug("Artifact successfully uploaded to artifactory");
+
+        // check scan status
+        boolean scanned;
+        do {
+            scanned = requestAndHandleScanStatus(context);
+        } while (!scanned);
+
+        // save reports from xray
+        if (artifactoryClient.requestScanReports()) {
+            // rewrite cycloneDX report with security report information
+            readAndConvertReports();
+        }
+
+        // delete artifact from artifactory
+        artifactoryClient.deleteArtifact();
+        artifactoryClient.deleteUploads();
+    }
+
+    private void readAndConvertReports() throws XrayWrapperException {
+        XrayWrapperReportData xrayWrapperReportData = new XrayWrapperReportData();
```

Comment:
Just a thought:
For simpler testing and maybe also for maintainance, extracting the 
read and convert reports method to its own class could have been helpful.

Also a return value or the context object as parameter could have been helpful - at least for testing.
It would be possible to check for the result file, compare it etc.
Also for the logic this could have been interesting - at least to log the report creation.

Another thing: IMO I would have maybe returned the report file as